package/MDAnalysis/analysis/psa.py

```diff
@@ -573,6 +573,14 @@ def dist_mat_to_vec(N, i, j):
     :Returns:
       int, index (of the matrix element) in the corresponding distance vector
     """
+
+    if not (isinstance(N, int) or isinstance(i, int) or isinstance(j, int)):
+        raise ValueError("N, i, j all must be of type int")
+
+    if i < 0 or j < 0 or N < 2:
+        error_str = "Matrix indices are invalid; i and j must be greater than 0 and N must be greater the 2"
+        raise ValueError(error_str)
+
     if i > N or j > N:
```

Comment:
According to @sseyler comment, this test should be `i > N - 1 or j > N`. Yet, because the function knows how to look at the lower triangle too, it should be `(j > i and (i > N - 1 or j > N)) or (j < i and (i > N or j > N - 1))`.